tests/windows_minidump_writer.rs

```diff
@@ -0,0 +1,140 @@
+#![cfg(all(target_os = "windows", target_arch = "x86_64"))]
+
+use minidump::{CrashReason, Minidump, MinidumpMemoryList, MinidumpSystemInfo, MinidumpThreadList};
+use minidump_writer::minidump_writer::MinidumpWriter;
+use std::mem;
+use windows_sys::Win32::{
+    Foundation::{EXCEPTION_ILLEGAL_INSTRUCTION, STATUS_INVALID_PARAMETER},
+    System::{
+        Diagnostics::Debug::{RtlCaptureContext, EXCEPTION_POINTERS, EXCEPTION_RECORD},
+        Threading::GetCurrentThreadId,
+    },
+};
+mod common;
+use common::start_child_and_return;
+
+fn get_crash_reason<'a, T: std::ops::Deref<Target = [u8]> + 'a>(
+    md: &Minidump<'a, T>,
+) -> CrashReason {
+    let exc: minidump::MinidumpException<'_> =
+        md.get_stream().expect("unable to find exception stream");
+
+    exc.get_crash_reason(
+        minidump::system_info::Os::Windows,
+        minidump::system_info::Cpu::X86_64,
+    )
+}
+
+/// Ensures that we can write minidumps for the current process, even if this is
+/// not necessarily the primary intended use case of out-of-process dumping
+#[test]
+fn dump_current_process() {
+    let mut tmpfile = tempfile::Builder::new()
+        .prefix("windows_current_process")
+        .tempfile()
+        .unwrap();
+
+    unsafe {
+        let mut exception_record: EXCEPTION_RECORD = mem::zeroed();
+        let mut exception_context = mem::MaybeUninit::uninit();
+
+        RtlCaptureContext(exception_context.as_mut_ptr());
```

Comment:
It might be worth asserting that this succeeds, since if it doesn't it will invoke UB in the next line

Reply:
`RtlCaptureContext` doesn't have a return value unfortunately.